Pinocchio can in fact compiled against urdfdom 0.2.3 
which is the version of Ubuntu 14.04

--- CMakeLists.txt~	2016-06-23 11:29:17.251249164 +0200
+++ CMakeLists.txt	2016-06-23 11:29:35.467476187 +0200
@@ -64,7 +64,7 @@
 ADD_REQUIRED_DEPENDENCY("eigen3 >= 3.0.5")
 ADD_OPTIONAL_DEPENDENCY("eigenpy >= 1.3.0")
 #ADD_OPTIONAL_DEPENDENCY("metapod >= 1.0.7")
-ADD_REQUIRED_DEPENDENCY("urdfdom >= 0.2.10")
+ADD_REQUIRED_DEPENDENCY("urdfdom >= 0.2.0")
 #ADD_OPTIONAL_DEPENDENCY("lua5.1")
 ADD_OPTIONAL_DEPENDENCY("hpp-fcl >= 0.4.1")
 IF(HPP_FCL_FOUND AND URDFDOM_FOUND)
